src/cryptoadvance/specter/server_endpoints/setup.py

```diff
@@ -0,0 +1,166 @@
+    setup_bitcoind_thread,
+    setup_bitcoind_directory_thread,
+)
+from ..util.tor_setup_tasks import (
+    setup_tor_thread,
+)
+
+rand = random.randint(0, 1e32)  # to force style refresh
+
+# Setup endpoint blueprint
+setup_endpoint = Blueprint("setup_endpoint", __name__)
+
+######################### Setup pages #######################################
+@setup_endpoint.route("/start/", methods=["GET"])
+@login_required
+def start():
+    app.specter.reset_setup("bitcoind")
+    app.specter.reset_setup("torbrowser")
+    return render_template("setup/start.jinja", specter=app.specter, rand=rand)
+
+
+@setup_endpoint.route("/tor/", methods=["GET"])
+@login_required
+def tor():
+    return render_template(
+        "setup/tor.jinja",
+        nextURL="setup_endpoint.node_type",
+        specter=app.specter,
+        rand=rand,
+    )
+
+
+@setup_endpoint.route("/node_type/", methods=["GET"])
+@login_required
+def node_type():
+    return render_template("setup/node_type.jinja", specter=app.specter, rand=rand)
+
+
+@setup_endpoint.route("/bitcoind/", methods=["GET"])
+@login_required
+def bitcoind():
+    return render_template("setup/bitcoind.jinja", specter=app.specter, rand=rand)
+
+
+@setup_endpoint.route("/bitcoind_datadir/", methods=["GET"])
+@login_required
+def bitcoind_datadir():
+    return render_template(
+        "setup/bitcoind_datadir.jinja", specter=app.specter, rand=rand
+    )
+
+
+@setup_endpoint.route("/end/", methods=["GET"])
+@login_required
+def end():
+    return render_template("setup/end.jinja", specter=app.specter, rand=rand)
+
+
+@setup_endpoint.route("/tor_setup/", methods=["GET"])
+@login_required
+def tor_from_settings():
+    return render_template(
+        "setup/tor.jinja",
+        nextURL="settings_endpoint.tor",
+        specter=app.specter,
+        rand=rand,
+    )
+
+
+######################### Background tasks #######################################
+@setup_endpoint.route("/setup_tor", methods=["POST"])
+@login_required
+def setup_tor():
+    if (
+        not os.path.isfile(app.specter.torbrowser_path)
+        and app.specter.setup_status["torbrowser"]["stage_progress"] == -1
+    ):
+        t = threading.Thread(target=setup_tor_thread, args=(app.specter,))
+        t.start()
+    elif os.path.isfile(app.specter.torbrowser_path):
+        return {"error": "Tor is already installed"}
+    elif app.specter.setup_status["torbrowser"]["stage_progress"] != -1:
+        return {"error": "Tor installation is still under progress"}
+    return {"success": "Starting Tor setup!"}
+
+
+@setup_endpoint.route("/setup_bitcoind", methods=["POST"])
+@login_required
+def setup_bitcoind():
+    app.specter.config["internal_node"]["datadir"] = request.form.get(
+        "bitcoin_core_datadir", app.specter.config["internal_node"]["datadir"]
+    )
+    app.specter._save()
+    if os.path.exists(app.specter.config["internal_node"]["datadir"]):
+        if request.form["override_data_folder"] != "true":
```

Comment:
I assumed that the folder gets completely purged BEFORE the new setup gets used. If that's not the case, someone with a broken chainstate or whatever broken state will never be able to recover without using "non-specter-tooling".

So imho we should have an else-clause which is purging that directory?!

Reply:
Added